File: content/problems/0204-count-primes/solutions/sieve-of-eratosthenes/solution.cpp
class Solution {
    public: int countPrimes(int n){
        vector<bool> composite(n);
        for(int p=2;(long long)p*p<n;++p)if(!composite[p])for(int value=p*p;value<n;value+=p)composite[value]=true;
        int count=0;
        for(int value=2;value<n;++value)if(!composite[value])++count;
        return count;
    }
};

File: content/problems/0204-count-primes/solutions/sieve-of-eratosthenes/Solution.java
class Solution {
    public int countPrimes(int n){
        boolean[] composite=new boolean[n];
        for(int p=2;(long)p*p<n;p++)if(!composite[p])for(int value=p*p;value<n;value+=p)composite[value]=true;
        int count=0;
        for(int value=2;value<n;value++)if(!composite[value])count++;
        return count;
    }
}
